googletest/include/gtest/gtest-printers.h

```diff
@@ -602,6 +604,22 @@ void PrintTo(const ::std::pair<T1, T2>& value, ::std::ostream* os) {
   *os << ')';
 }
 
+#if GTEST_HAS_RTTI
```

Comment:
@kuzkry I just added this `#if GTEST_HAS_RTTI` here, but now I realize it may not be necessary, as this code section does not use the `typeid` language feature. It only uses `std::type_info` and `std::type_index`, which appears acceptable, even with `-fno-rtti`. What do you think, shall I remove the `#if GTEST_HAS_RTTI` at this particular line of code from the pull request again?

Reply:
As I wrote, I'm indifferent here :) But it's good you handled `GTEST_HAS_RTTI` in the unit test file :)

Reply:
> I'm indifferent here :) But it's good you handled `GTEST_HAS_RTTI` in the unit test file :)

@kuzkry Thanks, I just removed that one `#if GTEST_HAS_RTTI` from gtest-printers.h, as I like to avoid depending on preprocessor macros whenever possible  😃 But I will certainly leave the `#if GTEST_HAS_RTTI` in the unit test file (which does use `typeid`).

BTW The `#include <typeinfo>` and `<typeindex>` compile well without RTTI, no need to `#if GTEST_HAS_RTTI` them either 😃